compiler/grainformat/reformat.re

```diff
@@ -0,0 +1,2753 @@
+            Doc.concat([
+              print_expression(
+                ~expr=expression,
+                ~parentIsArrow=false,
+                ~endChar=None,
+                ~original_source,
+                ~parent_loc,
+              ),
+              Doc.text(" = "),
+              print_expression(
+                ~expr=expression1,
+                ~parentIsArrow=false,
+                ~endChar=None,
+                ~original_source,
+                ~parent_loc,
+              ),
+            ])
+          };
+        } else {
+          Doc.concat([
+            print_expression(
+              ~expr=expression,
+              ~parentIsArrow=false,
+              ~endChar=None,
+              ~original_source,
+              ~parent_loc,
+            ),
+            Doc.text(" = "),
+            print_expression(
+              ~expr=expression1,
+              ~parentIsArrow=false,
+              ~endChar=None,
+              ~original_source,
+              ~parent_loc,
+            ),
+          ]);
+        };
+
+      | _ =>
+        Doc.concat([
+          print_expression(
+            ~expr=expression,
+            ~parentIsArrow=false,
+            ~endChar=None,
+            ~original_source,
+            ~parent_loc,
+          ),
+          Doc.text(" = "),
+          print_expression(
+            ~expr=expression1,
+            ~parentIsArrow=false,
+            ~endChar=None,
+            ~original_source,
+            ~parent_loc,
+          ),
+        ])
+      }
+
+    | /** Used for modules without body expressions */ PExpNull => Doc.nil
+    };
+
+  let endingDoc =
+    switch (endChar) {
+    | None => Doc.nil
+    | Some(d) => Doc.concat([d])
+    };
+
+  if (trailing_comment_docs == Doc.nil) {
+    Doc.concat([leading_comment_docs, expression_doc, endingDoc]);
+  } else {
+    Doc.concat([
+      leading_comment_docs,
+      expression_doc,
+      endingDoc,
+      trailing_comment_docs,
+    ]);
+  };
+}
+and value_bind_print =
+    (
+      export_flag: Asttypes.export_flag,
+      rec_flag,
+      mut_flag,
+      vbs: list(Parsetree.value_binding),
+      parent_loc: Grain_parsing__Location.t,
+      original_source: list(string),
+    ) => {
+  let exported =
+    switch (export_flag) {
+    | Nonexported => Doc.nil
+    | Exported => Doc.text("export ")
+    };
+  let recursive =
+    switch (rec_flag) {
+    | Nonrecursive => Doc.nil
+    | Recursive => Doc.text("rec ")
+    };
+  let mutble =
+    switch (mut_flag) {
+    | Immutable => Doc.nil
+    | Mutable => Doc.text("mut ")
+    };
+
+  let value_bindings =
+    List.map(
+      (vb: Parsetree.value_binding) => {
+        let expression =
+          print_expression(
+            ~expr=vb.pvb_expr,
+            ~parentIsArrow=false,
+            ~endChar=None,
+            ~original_source,
+            ~parent_loc,
+          );
+
+        // fix needed here?, we may  need to work out if all arguments fit on the line or will break
```

Comment:
```suggestion
        // TODO: fix needed here? we may need to work out if all arguments fit on the line or will break
```

Probably also good to open an issue.

Reply:
Reformatting of numbers is a good question re leading spaces. I  think this happens as they'll have been handled by the lexer